qml/Components/WallpaperResolver.qml

```diff
@@ -23,23 +24,35 @@ Item {
     // Provide a list of wallpapers to resolve here, preferred ones first
     property var candidates: []
 
+    property bool cache: true
+
     readonly property url background: {
         for (var i = 0; i < repeater.count; i++) {
-            if (repeater.itemAt(i).status === Image.Ready)
+            var image = repeater.itemAt(i);
+            if (image === null)
```

Comment:
does this actually happen? It seems an impossible case.

Reply:
This may have been a relic from when I was trying to use `root.candidates.slice(0, -1)` instead of `repeater.count` for this for loop. That actually caused some unexpected behavior in QML, the binding for `background` stopped firing when objects inside the repeater changed. I went back to `repeater.count` and it fixed the issue.

I'll fix it up.